Add try except around Redis set to handle corrupt redis data

kentsanggds opened the pull request:
## What

`try / except` added to catch any errors from corrupt Redis data in order to prevents bad data from stopping harvest processing and log the error for investigation, full stack trace output as part of log.

## Running tests for review 

Run the nose test in the ckan dev vm - 

`nosetests --with-pylons=test-core.ini ckanext/harvest/tests/ --nologcapture`

## Reference 

https://trello.com/c/MIA4o7ro/946-investigate-and-fix-corrupted-redis-data-issue

kentsanggds committed:
Add try except around Redis set to handle corrupt redis data

- prevents bad data from stopping harvest processing
- log the error for investigation, full stack trace output as part of log

thomasleese commented on ckanext/harvest/queue.py:
```diff
@@ -211,8 +211,14 @@ def __init__(self, redis, routing_key):
+            try:
+                self.redis.set(self.persistance_key(body),
+                            str(datetime.datetime.now()))
+            except Exception:
```
We can just have `except:` here since we don't care the type of exception.

kentsanggds replied:
👍

kentsanggds replied:
We decided to remove the stack trace as the exception is raised as an indicator that the redis data is corrupt.

thomasleese commented on ckanext/harvest/queue.py:
```diff
@@ -211,8 +211,14 @@ def __init__(self, redis, routing_key):
-                           str(datetime.datetime.now()))
+            try:
+                self.redis.set(self.persistance_key(body),
+                            str(datetime.datetime.now()))
```
It would be good to get the second argument to line up with the first.

kentsanggds replied:
Think it was, will reformat the code.

kentsanggds committed:
Update Redis exception to drop stack trace

Full stack trace is probably not needed as the Redis data corruption is probably happening somewhere else, the error log should make it easier to investigate it when it does happen.
